check_no_globals.go

```diff
@@ -18,6 +20,68 @@ type allowedExpression struct {
 	SelName string
 }
 
+// reports is the reports of found global variables that's not valid and will be
+// reported to the analysis pass.
+type report struct {
+	name string
+	pos  token.Pos
+}
+
+// Analyzer is the analasys analyzer for gochecknoglobals. Ironically enough,
+// this is in fact a global variable.
+var Analyzer = &analysis.Analyzer{ //nolint
+	Name:             "gochecknoglobals",
+	Doc:              "Don't allow global variables",
+	Run:              run,
+	Flags:            flags(),
+	Requires:         []*analysis.Analyzer{nolint.Analyzer},
```

Comment:
You're right. I saw this as a convenience since almost all linters I've ever used supports this (e.g. `pylint`, `shellcheck`, `rubocop`, `eslint` etc etc). But I guess this linter itself isn't really comparable to any of those. If you're running this as a part for your CI you're probably/hopefully running it through `golangci-lint` which would allow you to ignore reports.

It's a bit ironic though that this will mean that we can't pass our own code since the analyser is a global var due to standards. I could create a method constructing the analyser but I've never seen this done before.

Maybe we should drop this, minimise dependencies and special treatment and just accept that the linter won't pass it's own code. I'm fine with that for now.

Let me know if you want me to hide the global var in some kind of dummy constructor to access the analyser!